cloud/store/recover.go

```diff
@@ -0,0 +1,183 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package store
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/raft"
+	raftbolt "github.com/hashicorp/raft-boltdb/v2"
+	"github.com/vmihailenco/msgpack"
+)
+
+// Peer represents information about a peer in the Raft cluster
+type Peer struct {
+	Address  string `msgpack:"Address" json:"address"`
+	ID       string `msgpack:"ID" json:"id"`
+	Suffrage int64  `msgpack:"Suffrage" json:"-"`
+	NonVoter bool   `msgpack:"-" json:"non_voter"`
+}
+
+// genPeersFileFromBolt checks if there was an existing raft.db and has persisted config
+// so it generates peers.jon file to be used for raft.RecoverCluster to recover the cluster on start.
+func (st *Store) genPeersFileFromBolt(raftDBPath, peerFilePath string) error {
+	if _, err := os.Stat(raftDBPath); err == nil {
+		bdb, err := raftbolt.NewBoltStore(raftDBPath)
+		if err != nil {
+			return err
```

Comment:
I think there would be value in doing `fmt.Errorf("XXX: %w", err)` where `XXX` is the "current step" in the multiple err return in the function. In case of failure in production it could help narrow down which step had a problem.

Reply:
if there is an error here it will be descriptive enough, usually that step shouldn't be erroring but would add it

Reply:
actually we shouldn't return an error here though